inference-engine/tests/functional/plugin/shared/include/single_layer_tests/detection_output.hpp

```diff
@@ -0,0 +1,541 @@
+                    }
+                }
+                std::sort(scoreIndexPairs.begin(), scoreIndexPairs.end(), SortScorePairDescend<std::pair<int, int>>);
+                scoreIndexPairs.resize(attrs.keep_top_k[0]);
+                std::map<int, std::vector<int>> newIndices;
+                for (int j = 0; j < scoreIndexPairs.size(); ++j) {
+                    int label = scoreIndexPairs[j].second.first;
+                    int idx = scoreIndexPairs[j].second.second;
+                    newIndices[label].push_back(idx);
+                }
+                allIndices.push_back(newIndices);
+                numKept += attrs.top_k;
+            } else {
+                allIndices.push_back(indices);
+                numKept += numDet;
+            }
+        }
+
+        std::vector<float> result(numImages * attrs.keep_top_k[0] * 7, 0);
+        int count = 0;
+        for (int i = 0; i < numImages; ++i) {
+            const std::map<int, std::vector<float>>& confScores = confPreds[i];
+            const LabelBBox& decodeBboxesImage = decodeBboxes[i];
+            for (auto it = allIndices[i].begin(); it != allIndices[i].end(); ++it) {
+                int label = it->first;
+                const std::vector<float>& scores = confScores.find(label)->second;
+                int loc_label = attrs.share_location ? -1 : label;
+                const std::vector<NormalizedBBox>& bboxes = decodeBboxesImage.find(loc_label)->second;
+                std::vector<int>& indices = it->second;
+                for (int j = 0; j < indices.size(); ++j) {
+                    int idx = indices[j];
+                    result[count * 7 + 0] = i;
+                    result[count * 7 + 1] = attrs.decrease_label_id ? (label - 1) : label;
+                    result[count * 7 + 2] = scores[idx];
+                    const NormalizedBBox& bbox = bboxes[idx];
+
+                    float xmin = bbox.xmin;
+                    float ymin = bbox.ymin;
+                    float xmax = bbox.xmax;
+                    float ymax = bbox.ymax;
+
+                    if (attrs.clip_after_nms) {
+                          xmin = (std::max)(0.0f, (std::min)(1.0f, xmin));
+                          ymin = (std::max)(0.0f, (std::min)(1.0f, ymin));
+                          xmax = (std::max)(0.0f, (std::min)(1.0f, xmax));
+                          ymax = (std::max)(0.0f, (std::min)(1.0f, ymax));
+                    }
+
+                    result[count * 7 + 3] = xmin;
+                    result[count * 7 + 4] = ymin;
+                    result[count * 7 + 5] = xmax;
+                    result[count * 7 + 6] = ymax;
+                    ++count;
+                }
+            }
+        }
+        if (count < numImages * attrs.keep_top_k[0]) {
+            result[count * 7 + 0] = -1;
+        }
+        return result;
+    }
+};
+
+using DetectionOutputAttributes = std::tuple<
+    size_t,             // numClasses
```

Comment:
As I can see, the `detection output layer` parameters are different to [spec](https://github.com/openvinotoolkit/openvino/blob/master/docs/ops/detection/DetectionOutput_1.md) Could you align test parameters with the spec?

Reply:
fixed